kostyor/rest_api.py

```diff
@@ -3,19 +3,39 @@
 from collections import defaultdict
 
 from flask import Flask, jsonify, redirect, request, url_for
+from flask_restful import Api
 from keystoneauth1 import exceptions
 from keystoneauth1 import session
 from keystoneauth1.identity import v2
 import six
 
-from kostyor.common import constants
+from kostyor.common import constants, exceptions as kostyor_exc
 from kostyor.inventory import discover
 from kostyor.inventory import upgrades
+from kostyor import resources
 
 from kostyor.db import api as db_api
 from kostyor.db.api import db_session
 
 app = Flask(__name__)
+api = Api(app)
+
+
+# Flask-RESTful extends each 404 respond with custom message about close
+# endpoints. It works pretty awful and unexpected when you explicitly
+# specifies error message. Let's turn off it so any error message become
+# predicted.
+app.config['ERROR_404_HELP'] = False
+
+# New API endpoints should be implemented via Flask-RESTful and added here.
+# Once old ones are reimplemented - we can get rid of them and even provide
+# some factory function to return Flask application.
+api.add_resource(resources.Clusters, '/clusters')
+api.add_resource(resources.Cluster, '/clusters/<cluster_id>')
+
+api.add_resource(resources.Upgrades, '/clusters/<cluster_id>/upgrades')
+api.add_resource(
+    resources.Upgrade, '/clusters/<cluster_id>/upgrades/<upgrade_id>')
```

Comment:
Hm... Considering that we use recent upgrade task tied for cluster, I wonder do we really need to have this endpoint? Maybe it'd be better to `PUT` actions onto `/clusters/<cluster_id>/upgrades` instead.

What do you think, folks?